feat(content): The Successors - Enter the Frogtopus Aristocracy (#10296)
endless-sky/endless-sky: data/successors/successor weapons.txt

# Copyright (c) 2024 by Daeridanii
#
# Endless Sky is free software: you can redistribute it and/or modify it under the
# terms of the GNU General Public License as published by the Free Software
# Foundation, either version 3 of the License, or (at your option) any later version.
#
# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A

> # PARTICULAR PURPOSE.  See the GNU General Public License for more details.

Reviewer: ```suggestion
# PARTICULAR PURPOSE. See the GNU General Public License for more details.
```

			"start frame" 0
			"no repeat"
		sound "bimodal"
		"hit effect" "seeker impact"
		"hit effect" "electron impact" 3
		"fire effect" "moonbeam fleck" 4
		"inaccuracy" 1
		"velocity" 30
		"lifetime" 15
		"range override" 900
		"burst count" 12
		"burst reload" 6
		"reload" 30
		"firing energy" 80
		"firing heat" 120
		# Because parent projectiles inherit the damage of their submunitions, we must subtract the total damage of all submunitions from the parent projectile to get the desired behavior (different damage profiles over different ranges).
		"shield damage" -94
		# 50 (parent) - 3*48 (child)
		# close: 50 shield damage
		# far: 144 shield damage
		"hull damage" 75
		# 135 (parent) - 3*20 (child)
		# close: 135 hull damage
		# far: 60 hull damage
		"hit force" 120
		"cluster"
		"submunition" "Bimodal Switch" 3
	description `Precise and deadly, the "Ojet" Bimodal Coilgun has been the High Houses' armament of choice since long before humans ever took to the stars. It was designed quite simply as a perfect weapon; it is said that by the time the empire of the Predecessors fell, they had seen no need to install any other weapon on their ships for over a thousand years.`

> 	description `The "Ojet" fires hyperdense packets of charged particles that disperse as they travel: up close the concentrated bolt pierces armor, while at a distance the ionic spray destabilizes shielding. Together, a ship equipped with these weapons can adjust its damage profile just by changing its engagement range, allowing it to be highly effective against any foe.`

Reviewer 2: ```suggestion
	description `	The "Ojet" fires hyperdense packets of charged particles that disperse as they travel: up close the concentrated bolt pierces armor, while at a distance the ionic spray destabilizes shielding. Together, a ship equipped with these weapons can adjust its damage profile just by changing its engagement range, allowing it to be highly effective against any foe.`
```

outfit `Bimodal Coilgun Turret`
	category "Turrets"
	cost 2950000
	thumbnail "outfit/bimodal coilgun turret"
	licenses
		"High Houses"
	"mass" 55
	"outfit space" -55
	"weapon capacity" -55
	"turret mounts" -1
	weapon
		"hardpoint sprite" "hardpoint/bimodal turret"
		"hardpoint offset" 9.
		sprite "projectile/bimodal"
			"frame rate" 5
			"start frame" 0
			"no repeat"
		sound "bimodal"
		"hit effect" "seeker impact"
		"hit effect" "electron impact" 3
		"fire effect" "moonbeam fleck" 4
		"inaccuracy" 1
		"velocity" 30
		"lifetime" 15
		"range override" 900
		"turret turn" 3.5
		"burst count" 24
		"burst reload" 3
		"reload" 15
		"firing energy" 80
		"firing heat" 120
		"firing force" 25
		"shield damage" -94
		"hull damage" 75
		"hit force" 120
		"cluster"
		"submunition" "Bimodal Switch" 3

> 	description `Successor warships have traditionally relied on fixed weaponry to deal with their foes, but the High Houses do produce a turreted platform for the "Ojet" Coilgun so that their larger and slower vessels can maintain their combat effectiveness against more manuverable targets. It mounts two of the weapons side by side, rigged to fire double the amount of shots per burst to keep them in sync with the fixed hardpoints.`

Reviewer 2: ```suggestion
	description `Successor warships have traditionally relied on fixed weaponry to deal with their foes, but the High Houses do produce a turreted platform for the "Ojet" Coilgun so that their larger and slower vessels can maintain their combat effectiveness against more maneuverable targets. It mounts two of the weapons side by side, rigged to fire double the amount of shots per burst to keep them in sync with the fixed hardpoints.`
```

outfit "Bimodal Switch"
	weapon
		sprite "projectile/bimodal switch"
			"random start frame"
			"frame rate" 20
		"hit effect" "ion rain impact" 3
		"inaccuracy" 10
		"velocity" 30
		"lifetime" 6
		"random lifetime" 3
		"shield damage" 48
		"hull damage" 20

outfit "Drag Cannon"
	category "Guns"
	cost 455000
	thumbnail "outfit/drag cannon"
	licenses
		"High Houses"
	"mass" 12
	"outfit space" -12
	"weapon capacity" -12
	"gun ports" -1
	weapon
		sound "drag cannon"
		"fire effect" "moonbeam fleck" 2
		"hit effect" "drag cannon impact" 3
		"velocity" 0
		"lifetime" 0
		"range override" 375
		"velocity override" 25
		"burst count" 6
		"burst reload" 6
		"reload" 45
		"firing energy" 30
		"firing heat" 55
		"cluster"
		"submunition" "Drag Projectile" 3
	description `The Successors have a historical affinity for projectile weaponry which persists to the modern day. The Drag Cannon fires a short-ranged, highly inaccurate blast of projectiles that not only deal substantial damage but also temporarily disrupt the target's engines.`

> 	description `While it is an inelegant design in comparison to the time-tested "Ojet" Coilgun, and consequently less appreciated, the Drag Cannon has found a place on the Successors' fast attack craft primarily as a tool for slowing down enemy vessels so that they can be engaged by a larger fleet.`

Reviewer 2: ```suggestion
	description `	While it is an inelegant design in comparison to the time-tested "Ojet" Coilgun, and consequently less appreciated, the Drag Cannon has found a place on the Successors' fast attack craft primarily as a tool for slowing down enemy vessels so that they can be engaged by a larger fleet.`
```

		"burst count" 9
		"burst reload" 1
		"reload" 3
		"firing energy" 11
		"firing heat" 15
		"shield damage" 10.5
		"hull damage" 12
		"piercing" 0.05
	description "The introduction of the Pulse Laser was a wake-up call for the more astute in the High Houses, as it was the first time in hundreds of years that weapons were being manufactured by groups other than themselves. Most Successors see the turreted version as a continuation of this democratization of violence, while the more paranoid claim that the High Houses have infiltrated production and intentionally sabotaged the design."

outfit "Overcharged Pulse Laser"
	category "Guns"
	cost 84000
	thumbnail "outfit/overcharged pulse laser"
	licenses
		"Successor"
	"mass" 12
	"outfit space" -12
	"weapon capacity" -12
	"gun ports" -1
	weapon
		sprite "projectile/successor pulse laser"
			"start frame" 2
			"frame rate" 8
			"rewind"
		sound "laser"
		"fire effect" "moonbeam fleck" 3
		"hit effect" "electron impact"
		"inaccuracy" 5.5
		"velocity" 490
		"lifetime" 1
		"burst count" 4
		"burst reload" 1
		"reload" 7
		"firing energy" 9.1
		"firing heat" 30.8
		"shield damage" 14
		"hull damage" 13.3
		"piercing" 0.05
		"cluster"

> 	description `Unable to access large quantities of the high-quality weaponry produced by the High Houses, the People's Houses have had to work with what's available to them: mostly the relatively low-powered pulse lasers produced independently in Successor space. This pulse laser has had its safeties bypassed and its optics upgraded, allowing it to do more damage to shields at a lower power cost in exchange for producing substantially more heat.`

Reviewer 2: ```suggestion
	description `Unable to access large quantities of the high-quality weaponry produced by the High Houses, the People's Houses have had to work with what is available to them: mostly the relatively low-powered pulse lasers produced independently in Successor space. This pulse laser has had its safeties bypassed and its optics upgraded, allowing it to do more damage to shields at a lower power cost in exchange for producing substantially more heat.`
```

	thumbnail "outfit/sjeja kinetic lance"
	"mass" 60
	"outfit space" -60
	"weapon capacity" -60
	"energy capacity" 1900
	"gun ports" -1
	"spike capacity" 20
	weapon
		sprite "projectile/sjeja kinetic lance"
			"frame rate" 4
			"random start frame"
		sound "typhoon"
		ammo "High-Velocity Spike"
		icon "icon/hv spike"
		"fire effect" "moonbeam fleck" 20
		"fire effect" "electron impact"
		"hit effect" "torpedo hit"
		"die effect" "tiny explosion"
		"cluster"
		"velocity" 320
		"lifetime" 8
		"reload" 160
		"firing force" 1200
		"firing energy" 1900
		"firing ion" 3.0
		"firing heat" 4000
		"shield damage" 1336
		"hull damage" 1920
		"piercing" .25
		"hit force" 3750
		"penetration count" 5
	description "The largest and most powerful of the Successors' modern weapons, this massive coilgun is elegant in its simplicity, cleanly perforating targets at extreme range. Because the bank of exotic electromagnets it employs requires an unusually high burst of power when firing, the weapon comes with integrated battery banks in order to supply it with a large enough reserve of electricity to function."

outfit "High-Velocity Spike"
	category "Ammunition"
	licenses
		"High Houses"

> 	cost 4000

Reviewer 2: That is _expensive_.
Should the cost be lowered, even slightly?

Author: I'm not particularly attached to this price, but right now the cost-per-second to fire the "Sjeja" is more or less in line with other secondary weapons at the same tier: 1500 credits/sec is the same as the Bunrodea Swarm Pod, 1440 for the Wanderer Thunderhead, etc.
I suppose it might make sense for this to be a little cheaper given it's not an explosive weapon.

Reviewer 2: Maybe down to 3500? Though I suppose this does do more damage per second then the other two, so it should be expensive.

	thumbnail "outfit/hv spike"
	"mass" .5
	"spike capacity" -1
	description `The projectiles fired by the "Sjeja" Kinetic Lance are too large to be synthesized on-the-fly from the ship's reserves, and must be carried separately.`

outfit `High-Velocity Spike Rack`
	category "Ammunition"
	cost 350000
	licenses
		"High Houses"
	thumbnail "outfit/hv spike rack"
	"mass" 7
	"outfit space" -12
	"energy capacity" 800
	"spike capacity" 16

> 	description `Though the built-in ammunition reserves of the "Sjeja" Kinetic Lance are sufficient for all but the most protracted engagements, the High Houses do manufacture this additional storage unit for use on their largest warships. The rack includes a small set of batteries to help support the high power costs of the "Sjeja."`

Reviewer 2: ```suggestion
	description `Though the built-in ammunition reserves of the "Sjeja" Kinetic Lance are sufficient for all but the most protracted engagements, the High Houses do manufacture this additional storage unit for use on their largest warships. The rack includes a small set of batteries to help support the high power costs of the "Sjeja."`

```

		"turret turn" 3.5
		"burst count" 24
		"burst reload" 3
		"reload" 15
		"firing energy" 80
		"firing heat" 120
		"firing force" 25
		"shield damage" -94
		"hull damage" 75
		"hit force" 120
		"cluster"
		"submunition" "Bimodal Switch" 3
	description `Successor warships have traditionally relied on fixed weaponry to deal with their foes, but the High Houses do produce a turreted platform for the "Ojet" Coilgun so that their larger and slower vessels can maintain their combat effectiveness against more maneuverable targets. It mounts two of the weapons side by side, rigged to fire double the amount of shots per burst to keep them in sync with the fixed hardpoints.`

effect "bimodal impact"
	sprite "effect/bimodal impact"
		"frame rate" 30
		"no repeat"
	lifetime 8
	"velocity scale" 0.2

outfit "Bimodal Switch"
	weapon
		sprite "projectile/bimodal switch"
			"random start frame"
			"frame rate" 20
		"hit effect" "bimodal switch impact"
		"inaccuracy" 10
		"velocity" 30
		"lifetime" 6
		"random lifetime" 3
		"shield damage" 48
		"hull damage" 20

effect "bimodal switch impact"
	sprite "effect/bimodal switch impact"
		"frame rate" 12
		"no repeat"
	lifetime 30
	"velocity scale" 0.02

> outfit "Drag Cannon"

Reviewer 3: The Drag Cannon has very high DPS for its size, applies a small amount of slowing damage, and benefits from a clustered burst firing mode that allows it to output large amounts of damage in a short space of time - all of which would make it a very strong weapon. On the other hand, the range is very low, and with 20 degrees of inaccuracy the effective damage to all but the largest targets at anything beyond point-blank range is much more reasonable.

		"range override" 375
		"velocity override" 25
		"burst count" 6
		"burst reload" 6
		"reload" 45
		"firing energy" 30
		"firing heat" 55
		"cluster"
		"submunition" "Drag Projectile" 3
	description `The Successors have a historical affinity for projectile weaponry which persists to the modern day. The Drag Cannon fires a short-ranged, highly inaccurate blast of projectiles that not only deal substantial damage but also temporarily disrupt the target's engines.`
	description `	While it is an inelegant design in comparison to the time-tested "Ojet" Coilgun, and consequently less appreciated, the Drag Cannon has found a place on the Successors' fast attack craft primarily as a tool for slowing down enemy vessels so that they can be engaged by a larger fleet.`

outfit "Drag Projectile"
	weapon
		sprite "projectile/drag cannon"
			"frame rate" 20
			"no repeat"
		"hit effect" "drag cannon impact"
		"inaccuracy" 20
		"velocity" 25
		"lifetime" 10
		"random lifetime" 5
		"slowing damage" 0.45
		"shield damage" 52
		"hull damage" 48

effect "drag cannon impact"
	sprite "effect/drag cannon impact"
		"frame rate" 10
		"no repeat"
	lifetime 20
	"velocity scale" 0
	"random velocity" 1

> outfit "Successor Pulse Laser"

Reviewer 3: Compared to the Thrasher Cannon, which is the same size and has a similar ratio of shield to hull damage, this deals about 10% less damage, but has a considerably longer range, is a beam and has a tiny amount of piercing. I don't think the piercing is a problem, and I think the tradeoff of reduced damage in exchange for more range is reasonable.

> outfit "Successor Pulse Laser"

Reviewer 2: Probably to avoid confusion with the Pulse Cannon of the UHai.

> outfit "Successor Pulse Laser"

Author: Also to distinguish it from the Guile Pulse Laser of the Gegno Scin.

	category "Guns"
	series "Guns: Successors"
	index 30
	cost 69000
	thumbnail "outfit/successor pulse laser"
	licenses
		"Successor"
	"mass" 10
	"outfit space" -10
	"weapon capacity" -10
	"gun ports" -1
	weapon
		sprite "projectile/successor pulse laser"
			"random start frame"
			"frame rate" 5
		sound "successor pulse laser"
		"hit effect" "successor pulse laser impact"
		"inaccuracy" 0.4
		"velocity" 490
		"lifetime" 1
		"burst count" 9
		"burst reload" 1
		"reload" 6
		"firing energy" 11
		"firing heat" 15
		"shield damage" 10.5
		"hull damage" 12
		"piercing" 0.05
	description "This shield-penetrating laser is one of the few weapons manufactured independently in Successor space, rather than under the auspices and restrictions of the High Houses. As a consequence of this, it is neither especially powerful nor efficient, but its low cost and space requirements make it a nonetheless relatively appealing defensive choice for those without access to the High Houses' more potent weaponry."

> outfit "Pulse Laser Turret"

Reviewer 3: Traverse seems very reasonable for a turret of this size. Other than that, see the comment above on the gun version.

	category "Turrets"
	series "Turrets: Successors"
	index 20
	cost 145000
	thumbnail "outfit/successor pulse laser turret"
	licenses
		"Successor"
	"mass" 30
	"outfit space" -30
	"weapon capacity" -30
	"turret mounts" -1
	weapon
		sprite "projectile/successor pulse laser"
			"random start frame"
			"frame rate" 5
		"hardpoint sprite" "hardpoint/successor pulse laser turret"
		"hardpoint offset" 8
		sound "successor pulse laser"
		"hit effect" "successor pulse laser impact"
		"inaccuracy" 0.4
		"velocity" 490
		"turret turn" 2.5
		"lifetime" 1
		"burst count" 9
		"burst reload" 1
		"reload" 3
		"firing energy" 11
		"firing heat" 15
		"shield damage" 10.5
		"hull damage" 12
		"piercing" 0.05

> 	description "The introduction of the Pulse Laser was a wake-up call for the more astute in the High Houses, as it was the first time in hundreds of years that weapons were being manufactured by groups other than themselves. Most Successors see the turreted version as a continuation of this democratization of violence, while the more paranoid claim that the High Houses have infiltrated production and intentionally sabotaged the design."

Reviewer 4: > democratization of violence

I love this.

effect "successor pulse laser impact"
	sprite "effect/successor pulse laser impact"
		"frame rate" 30
		"random start frame"
	lifetime 3
	"velocity scale" 0

> outfit "Overcharged Pulse Laser"

Reviewer 3: I would consider this to be a noticeably weaker outfit than the standard pulse laser. Its dps/ton is slightly worse, it's slightly easier to support, and having higher shield damage but lower hull damage makes the piercing less useful. The firing cycle is too short to really take advantage of the burst fire with cluster firing mode, too.

	category "Guns"
	series "Guns: Successors"
	index 40
	cost 84000
	thumbnail "outfit/overcharged pulse laser"
	licenses
		"Successor"
	"mass" 12
	"outfit space" -12
	"weapon capacity" -12
	"gun ports" -1
	weapon
		sprite "projectile/successor pulse laser"
			"start frame" 2
			"frame rate" 8
			"rewind"
		sound "overcharged pulse laser"
		"fire effect" "moonbeam fleck" 3
		"hit effect" "overcharged pulse laser impact"
		"inaccuracy" 5.5
		"velocity" 490
		"lifetime" 1
		"burst count" 4
		"burst reload" 1
		"reload" 7
		"firing energy" 9.1
		"firing heat" 30.8
		"shield damage" 14
		"hull damage" 13.3
		"piercing" 0.05
		"cluster"
	description `Unable to access large quantities of the high-quality weaponry produced by the High Houses, the People's Houses have had to work with what is available to them: mostly the relatively low-powered pulse lasers produced independently in Successor space. This pulse laser has had its safeties bypassed and its optics upgraded, allowing it to do more damage to shields at a lower power cost in exchange for producing substantially more heat.`

> outfit "Overcharged Laser Turret"

Reviewer 3: Similarly to the gun version, this is a noticeably less powerful weapon than the standard pulse laser turret. The high traverse is reasonable for such a small weapon.

	"weapon capacity" -24
	"turret mounts" -1
	weapon
		sprite "projectile/successor pulse laser"
			"start frame" 2
			"frame rate" 8
			"rewind"
		"hardpoint sprite" "hardpoint/overcharged pulse laser turret"
		"hardpoint offset" 8
		sound "overcharged pulse laser"
		"fire effect" "moonbeam fleck" 3
		"hit effect" "overcharged pulse laser impact"
		"inaccuracy" 5.5
		"velocity" 490
		"lifetime" 1
		"turret turn" 3.5
		"burst count" 4
		"burst reload" 1
		"reload" 7
		"firing energy" 12.6
		"firing heat" 43.05
		"shield damage" 19.6
		"hull damage" 18.667
		"piercing" 0.05
		"cluster"
	description `Hand-crafted by the more colorful members of the People's Houses, this bespoke overcharged Pulse Laser turret has had even more power pumped into its laser generator compared to the fixed version, producing a truly prodigious amount of heat for a weapon of its size. Despite its inelegant design, it is nonetheless a fearsome weapon, especially considering that it retains the ability to partially punch through shielding.`

effect "overcharged pulse laser impact"
	sprite "effect/overcharged pulse laser impact"
		"frame rate" 30
		"random start frame"
	lifetime 6
	"velocity scale" 0

> outfit `"Sjeja" Kinetic Lance`

Reviewer 3: This is both nontrivial to support energy-wise, requiring nearly the full output of a Yellow Sun or Epoch Cell, and ammo-limited. In exchange for that, it can deal major damage, with almost no reaction time, at very long range, and with piercing and penetration.
In the player's hands, I think it's balanced out by the limitation of requiring ammunition. Its effectiveness is reasonably in-line with some other high-tier missile weapons - its DPS is slightly below that of the T2 Firestorm, and considerably below that of the T2.5 Finisher Pod, while the weapon itself is larger and hugely more power-hungry than either. Getting to and from Successor space is nontrivial too, so sourcing ammunition is more difficult than for many other missiles.

My main concern is the lack of options for the player if they end up on the recieving end of the Sjeja in future story content. Since it reaches its full range in only 8 frames, there's not really scope for dodging it, and at that kind of long range circling round faster than the opponent can turn isn't really an option either (even if the firing ship has an extremely low turn rate of only 40 degrees/sec, at 1900 range you'd need a speed of over 1200 to stay ahead). The high hit force and firing force make closing the distance difficult too, particularly in lighter ships that are pushed more.

	licenses
		"High Houses"
	thumbnail "outfit/sjeja kinetic lance"
	"mass" 60
	"outfit space" -60
	"weapon capacity" -60
	"energy capacity" 1900
	"gun ports" -1
	"spike capacity" 20
	weapon
		sprite "projectile/sjeja kinetic lance"
			"frame rate" 4
			"random start frame"
		sound "typhoon"
		ammo "High-Velocity Spike"
		icon "icon/hv spike"
		"fire effect" "moonbeam fleck" 20
		"fire effect" "electron impact"
		"hit effect" "torpedo hit"
		"die effect" "tiny explosion"
		"cluster"
		"velocity" 320
		"lifetime" 8
		"reload" 160
		"firing force" 800
		"firing energy" 1900
		"firing ion" 3.0
		"firing heat" 4000
		"shield damage" 1336
		"hull damage" 1920
		"piercing" 0.15
		"hit force" 800
		"penetration count" 5
	description "The largest and most powerful of the Successors' modern weapons, this massive superconducting coilgun is elegant in its simplicity, cleanly perforating targets at extreme range. Although the weapon requires a substantial burst of power in order to fire, it is capable of storing some of this within the coils themselves, allowing it to be fired without additional battery banks at the cost of inflicting a lingering power draw on the ship's systems."

> outfit "High-Velocity Spike"

Author: Many alien ammo outfits don't seem to have an additional generic description, so I'm inclined to avoid adding one here for the sake of brevity.
